Tests/FluentAssertions.Specs/Equivalency/BasicEquivalencySpecs.cs

```diff
@@ -3462,6 +3462,546 @@ public class OuterWithObject
+            {
+                // Arrange
+                var subject = new
+                {
+                    Parent = new[]
+                    {
+                        new SubjectWithProperty1
+                        {
+                            Property1 = "Hello"
+                        }
+                    }
+                };
+
+                var expectation = new
+                {
+                    Parent = new[]
+                    {
+                        new ExpectationWithProperty2
+                        {
+                            Property2 = "Hello"
+                        }
+                    }
+                };
+
+                // Act / Assert
+                subject.Should()
+                    .BeEquivalentTo(expectation, opt => opt
+                        .WithMapping("Parent[].Property2", "Parent[].Property1"));
+            }
+
+            [Fact]
+            public void Properties_can_be_mapped_by_name()
+            {
+                // Arrange
+                var subject = new SubjectWithProperty1
+                {
+                    Property1 = "Hello"
+                };
+
+                var expectation = new ExpectationWithProperty2
+                {
+                    Property2 = "Hello"
+                };
+
+                // Act / Assert
+                subject.Should()
+                    .BeEquivalentTo(expectation, opt => opt
+                    .WithMapping("Property2", "Property1"));
+            }
+
+            [Fact]
+            public void Fields_can_be_mapped_by_name()
+            {
+                // Arrange
+                var subject = new ClassWithSomeFieldsAndProperties
+                {
+                    Field1 = "Hello",
+                    Field2 = "John"
+                };
+
+                var expectation = new ClassWithSomeFieldsAndProperties
+                {
+                    Field1 = "John",
+                    Field2 = "Hello"
+                };
+
+                // Act / Assert
+                subject.Should()
+                    .BeEquivalentTo(expectation, opt => opt
+                    .WithMapping("Field2", "Field1")
+                    .WithMapping("Field1", "Field2"));
+            }
+
+            [Fact]
+            public void Fields_can_be_mapped_to_a_property_by_name()
+            {
+                // Arrange
+                var subject = new ClassWithSomeFieldsAndProperties
+                {
+                    Property1 = "John"
+                };
+
+                var expectation = new ClassWithSomeFieldsAndProperties
+                {
+                    Field1 = "John",
+                };
+
+                // Act / Assert
+                subject.Should()
+                    .BeEquivalentTo(expectation, opt => opt
+                        .WithMapping("Field1", "Property1")
+                        .Including(e => e.Field1));
```

Comment:
Why is `.Including(e => e.Field1)` necessary?

Reply:
Because I was too lazy to create another type with just one property and one field.

Reply:
Pheew, I thought it was some requirement to make the field<->property mapping actually work.